src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureFilterTypes.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using System.Text;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement
+{
+    /// <summary>
+    /// Types of built-in feature filters.
+    /// </summary>
+    internal class FeatureFilterTypes
+    {
+        private const string CustomFilter = "Custom";
+        private const string PercentageFilter = "Percentage";
+        private const string TimeWindowFilter = "Time";
+        private const string TargetingFilter = "Targeting";
+        private const string FilterTypeDelimiter = "+";
+
+        public bool UsesCustomFilter { get; set; } = false;
+        public bool UsesPercentageFilter { get; set; } = false;
+        public bool UsesTimeWindowFilter { get; set; } = false;
+        public bool UsesTargetingFilter { get; set; } = false;
+       
+        public bool UsesAnyFeatureFilter()
+        {
+            return UsesCustomFilter || UsesPercentageFilter || UsesTimeWindowFilter || UsesTargetingFilter;
+        }
+
+        public void ResetFeatureFilters()
+        {
+            UsesCustomFilter = false;
+            UsesPercentageFilter = false;
+            UsesTimeWindowFilter = false;
+            UsesTargetingFilter = false;
+        }
+
+        public override string ToString()
```

Comment:
The knowledge of this string is specific to the telemetry, I'd like to keep the implementation close by there if possible.

Reply:
I combined all telemetry operations in a single class called FeatureFilterTelemetry.